web/src/features/chat/components/ChatMessage.tsx

```
@@ -152,15 +159,18 @@ export const ChatMessage: React.FC<ChatMessageProps> = ({
 
     React.useEffect(() => {
-        if ((status === 'building' || status === 'done') && !hasAutoCollapsedRef.current) {
+        if (
+            (status === 'building' || status === 'done' || Boolean(plan)) &&
+            !hasAutoCollapsedRef.current
+        ) {
             setIsThoughtsOpen(false)
             hasAutoCollapsedRef.current = true
         }
-        if (status === 'thinking') {
+        if (status === 'thinking' && !Boolean(plan)) {
```

ESLint

[error] Redundant Boolean call.

(no-extra-boolean-cast)